lib/cylc/network/https/client_reporter.py

```diff
@@ -131,19 +131,22 @@ def signout(self, server_obj):
                 caller.user, caller.host, caller.prog_name, caller.uuid))
         try:
             del self.clients[caller.uuid]
-        except:
+        except KeyError:
             # Already forgotten.
             pass
         self._housekeep()
 
     def _housekeep(self):
         """Forget inactive clients."""
 
-        for uuid in self.clients.keys():
+        for uuid in self.clients.copy():
```

Comment:
But doesn't `dict.keys()` return a list, so deletion of list elements does not affect the original source dict?